frida-gum/src/error.rs

```diff
@@ -3,32 +3,37 @@
  *
  * Licence: wxWindows Library Licence, Version 3.1
  */
-
-use thiserror::Error;
+use core::fmt;
 
 /// Custom `Error` for Frida
-#[derive(Error, Debug)]
 pub enum Error {
     /// Bad signature during Interceptor operation
-    #[error("Bad signature")]
     InterceptorBadSignature,
 
     /// Function is already replaced during Interceptor operation
-    #[error("Function already replaced")]
     InterceptorAlreadyReplaced,
 
     /// Policy violation
-    #[error("Policy violation")]
     PolicyViolation,
 
     /// Other Interceptor error
-    #[error("Interceptor error")]
     InterceptorError,
 
     /// Memory access error
-    #[error("Memory access error")]
     MemoryAccessError,
 
-    #[error("Wrong type")]
     WrongType,
 }
+
+impl fmt::Debug for Error {
```

Comment:
One more thought here... we don't want Debug, we actually want Display. `thiserror` actually implements both, but I think we can get away with just Display.

Reply:
I moved to Display instead, and this resulted in some errors where `NativePointer`s were being wrongly coerced into strings. e.g. the string was being read and in the event of a memory error, the resulting structure would have it's field set to the error message. Since the functions didn't return an error, I instead replaced the `unwrap` with `unwrap_or_default`. Although I'm happy if you want to change to something else?

This resolved the issues for the modules, but the tests were still relying on `Debug` since they were simply using `unwrap` and presumably `Debug` support is required for the resulting panic. I tackled this by implementing this as a trait and having the `Debug` implementation simply call the one for `Display`. I could have simply added a `#[derive(Debug)]` but thought this might look better. Again happy if you would rather change. I added this as a separate commit too, so that you can pick and choose what you want.

Reply:
Thanks for the feedback, hopefully these changes should tackle it. I'm not precious about my code, I'm am perfectly happy if you want to tweak or change any of my commits before you merge rather than review if it makes things easier for you, particularly if it is just the finer details like this.

Thanks again.

Reply:
Awesome. Thanks very much.

Reply:
Will you be able to make a new dot release containing the changes? Or do you have a particular release schedule?

Reply:
Done in v0.11.0

Reply:
Great. Thanks.